modules/cuckooml/cuckooml.py

```diff
@@ -0,0 +1,1677 @@
+# Copyright (C) 2010-2013 Claudio Guarnieri.
+# Copyright (C) 2014-2016 Cuckoo Foundation.
+# This file is part of Cuckoo Sandbox - http://www.cuckoosandbox.org
+# See the file 'docs/LICENSE' for copying permission.
+
+import collections
+import datetime
+import itertools
+import json
+import os
+import re
+import sys
+import time
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from hdbscan import HDBSCAN
```

Comment:
These imported library will have to be added to `requirements.txt` so that users will install them and may actually use this functionality. Also, `try` / `except ImportError` statements should be placed here in case these libraries have not been installed (due to their nature, this is quite likely).

In the future, through the Python `package` (see also #863), we may be able to force these packages on to users, if not by explicitly enabling the `machine learning` capabilities.
That said, are there any special OS package requirements when installing these Python libraries? E.g., some libraries that compile native Python extensions require the `libpython-dev` package in Ubuntu.